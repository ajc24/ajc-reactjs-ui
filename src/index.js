export { default as Footer } from './ui-elements/Footer';
export { default as Header } from './ui-elements/Header';
export { default as HeaderLogo } from './ui-elements/HeaderLogo';
export { default as HeaderTitleAndSubtitle } from './ui-elements/HeaderTitleAndSubtitle';
export { default as MainContentPageTitle } from './ui-elements/MainContentPageTitle';
export { default as MainContentPanel } from './ui-elements/MainContentPanel';
export { default as MainContentText } from './ui-elements/MainContentText';
export { default as Menu } from './ui-elements/Menu';
export { default as Screenshot } from './ui-elements/Screenshot';
export { default as SmallMenuIcon } from './ui-elements/SmallMenuIcon';
export { default as SmallRightArrowIcon } from './ui-elements/SmallRightArrowIcon';
