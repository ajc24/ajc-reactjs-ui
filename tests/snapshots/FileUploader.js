import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-jest-enzyme';
import { FileUploader } from '../../src';

describe('File Uploader', () => {
  let componentDidMountSpy;

  beforeAll(() => {
    componentDidMountSpy = jest
      .spyOn(FileUploader.prototype, 'componentDidMount')
      .mockImplementation(() => {});
  });

  afterAll(() => {
    componentDidMountSpy.mockRestore();
  });

  describe('Default props and rendering', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <FileUploader id="default-file-uploader-id" />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - File upload with red buttons, custom title text and additional upper spacing', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <FileUploader id="custom-file-uploader-id" addUpperSpacing={true} colour="red" labelText="Custom File Uploader Title:" />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });
});
