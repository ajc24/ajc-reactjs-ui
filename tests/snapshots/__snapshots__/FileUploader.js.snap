// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File Uploader Default props and rendering verifies the snapshot for the component 1`] = `
<div
  className="ajc-file-uploader-container ajc-default"
>
  <p
    className="ajc-text-paragraph-container ajc-text-black ajc-default"
  >
    <b
      className="ajc-text-paragraph-heading"
    >
      Upload a File:
    </b>
  </p>
  <div
    className="ajc-buttons-list-container ajc-buttons-list-container-left-align ajc-default"
  >
    <button
      className="ajc-button ajc-default ajc-button-grey ajc-button-right-spacing"
      id="default-file-uploader-id-choose-button"
      onClick={[Function]}
      type="button"
    >
      Choose a file to upload
    </button>
    <button
      className="ajc-button ajc-default ajc-button-grey"
      id="default-file-uploader-id-clear-button"
      onClick={[Function]}
      type="button"
    >
      Clear selection
    </button>
  </div>
  <p
    className="ajc-text-paragraph-container ajc-text-black ajc-default"
  >
    <i>
      <span
        id="default-file-uploader-id-file-status"
      />
    </i>
  </p>
  <label
    className="ajc-file-uploader-input-label"
    for="default-file-uploader-id-file"
  >
    Hidden File Input:
  </label>
  <input
    className="ajc-file-uploader-input"
    id="default-file-uploader-id-file"
    name="default-file-uploader-id-file"
    onChange={[Function]}
    type="file"
  />
</div>
`;

exports[`File Uploader Transferred props and rendering - File upload with red buttons, custom title text and additional upper spacing verifies the snapshot for the component 1`] = `
<div
  className="ajc-file-uploader-container ajc-default"
>
  <p
    className="ajc-text-paragraph-container ajc-text-black ajc-default ajc-text-paragraph-upper-spacing"
  >
    <b
      className="ajc-text-paragraph-heading"
    >
      Custom File Uploader Title:
    </b>
  </p>
  <div
    className="ajc-buttons-list-container ajc-buttons-list-container-left-align ajc-default"
  >
    <button
      className="ajc-button ajc-default ajc-button-red ajc-button-right-spacing"
      id="custom-file-uploader-id-choose-button"
      onClick={[Function]}
      type="button"
    >
      Choose a file to upload
    </button>
    <button
      className="ajc-button ajc-default ajc-button-red"
      id="custom-file-uploader-id-clear-button"
      onClick={[Function]}
      type="button"
    >
      Clear selection
    </button>
  </div>
  <p
    className="ajc-text-paragraph-container ajc-text-black ajc-default"
  >
    <i>
      <span
        id="custom-file-uploader-id-file-status"
      />
    </i>
  </p>
  <label
    className="ajc-file-uploader-input-label"
    for="custom-file-uploader-id-file"
  >
    Hidden File Input:
  </label>
  <input
    className="ajc-file-uploader-input"
    id="custom-file-uploader-id-file"
    name="custom-file-uploader-id-file"
    onChange={[Function]}
    type="file"
  />
</div>
`;
